activesupport/lib/active_support/inflector/methods.rb

```diff
@@ -207,7 +207,13 @@ def classify(table_name)
     #
     #   dasherize('puni_puni') # => "puni-puni"
     def dasherize(underscored_word)
-      underscored_word.tr("_".freeze, "-".freeze)
+      @_dasherize_cache ||= {}
+      @_dasherize_cache[underscored_word] ||=
+        if underscored_word.include?("_")
```

Comment:
@rafaelfranca @jeremy thoughts on this change separate from the caching?

Reply:
Looks fine to me.

Reply:
Did some profiling on this and in the scenario with no underscore it provides a speedup but in the scenario with an underscore it was equivalently slower - which doesn't seem worth it to me.